### PR TITLE
[NRLF-663] Add Open Source narrative

### PR DESCRIPTION
**Pull request by nomad3k**

### CODE REVIEW

File: swagger/producer-static/narrative.yaml [hunk 1]
```diff
@@ -127,6 +127,32 @@ info:
...
+    |---------------------------|-----------------------------------------------------------|--------------------------------------------------------------------------------------------------|
+    | NRL v3 API                | Source code for the core API and sandbox                  | [GitHub repo](https://github.com/NHSDigital/nrlf)                                                |
+    | NRL v3 Producer API       | Source code for the producer API proxy and specification. | [GitHub repo](https://github.com/NHSDigital/nrl-producer-api)                                    |
+    | NRL v3 Consumer API       | Source code for the consumer API proxy and specification. | [GitHub repo](https://github.com/NHSDigital/nrl-consumer-api)                                    |
+    | Lambda Pipeline Package   | Source code for the lambda pipeline utility               | [GitHub repo](https://github.com/NHSDigital/nrlf-lambda-pipeline) |
+    | Lambda Logging Package    | Source code for the lambda logging utility                | [GitHub repo](https://github.com/NHSDigital/nrlf-lambda-logging)
```
Comment (axelkrastek1-nhs) [inline swagger/producer-static/narrative.yaml:150]: The table is missing some pipes at the end and aligned.

File: swagger/consumer-static/narrative.yaml [hunk 2]
```diff
@@ -137,6 +138,32 @@ info:
...
+
+    | Resource                  | Description                                               | Links                                                                      |
+    |---------------------------|-----------------------------------------------------------|----------------------------------------------------------------------------|
+    | NRL v3 API                | Source code for the core API and sandbox                  | [GitHub repo](https://github.com/NHSDigital/nrlf)                          |
+    | NRL v3 Producer API       | Source code for the producer API proxy and specification. | [GitHub repo](https://github.com/NHSDigital/nrl-producer-api)              |
+    | NRL v3 Consumer API       | Source code for the consumer API proxy and specification. | [GitHub repo](https://github.com/NHSDigital/nrl-consumer-api)              |
+    | Lambda Pipeline Package   | Source code for the lambda pipeline utility               | [GitHub repo](https://github.com/NHSDigital/nrlf-lambda-pipeline) |
+    | Lambda Logging Package    | Source code for the lambda logging utility                | [GitHub repo](https://github.com/NHSDigital/nrlf-lambda-logging)
```
Comment (axelkrastek1-nhs) [inline swagger/consumer-static/narrative.yaml:161]: The table is missing some pipes at the end and aligned.

File: api/producer/swagger.yaml [hunk 3]
```diff
@@ -111,6 +111,20 @@ info:
...
 
+    ### Open source
+
+    You might find the following open source resources useful:
+
+    | Resource                  | Description                                               | Links                                                                      |
+    |---------------------------|-----------------------------------------------------------|----------------------------------------------------------------------------|
+    | NRL v3 API                | Source code for the core API and sandbox                  | [GitHub repo](https://github.com/NHSDigital/nrlf)                          |
+    | NRL v3 Producer API       | Source code for the producer API proxy and specification. | [GitHub repo](https://github.com/NHSDigital/nrl-producer-api)              |
+    | NRL v3 Consumer API       | Source code for the consumer API proxy and specification. | [GitHub repo](https://github.com/NHSDigital/nrl-consumer-api)              |
```
Comment (axelkrastek1-nhs) [inline api/producer/swagger.yaml:122]: This is missing lambda packaging and lambda logging here and the errors are still below the open source heading.

File: api/producer/record-locator/producer.yaml [hunk 4]
```diff
@@ -111,6 +111,20 @@ info:
...
 
+    ### Open source
+
+    You might find the following open source resources useful:
+
+    | Resource                  | Description                                               | Links                                                                      |
+    |---------------------------|-----------------------------------------------------------|----------------------------------------------------------------------------|
+    | NRL v3 API                | Source code for the core API and sandbox                  | [GitHub repo](https://github.com/NHSDigital/nrlf)                          |
+    | NRL v3 Producer API       | Source code for the producer API proxy and specification. | [GitHub repo](https://github.com/NHSDigital/nrl-producer-api)              |
+    | NRL v3 Consumer API       | Source code for the consumer API proxy and specification. | [GitHub repo](https://github.com/NHSDigital/nrl-consumer-api)              |
```
Comment (axelkrastek1-nhs) [inline api/producer/record-locator/producer.yaml:122]: This is missing lambda packaging and lambda logging here and the errors are still below the open source heading.

File: api/consumer/swagger.yaml [hunk 5]
```diff
@@ -122,7 +122,22 @@ info:
...
 
+    ### Open source
+
+    You might find the following open source resources useful:
+
+    | Resource                  | Description                                               | Links                                                                      |
+    |---------------------------|-----------------------------------------------------------|----------------------------------------------------------------------------|
+    | NRL v3 API                | Source code for the core API and sandbox                  | [GitHub repo](https://github.com/NHSDigital/nrlf)                          |
+    | NRL v3 Producer API       | Source code for the producer API proxy and specification. | [GitHub repo](https://github.com/NHSDigital/nrl-producer-api)              |
+    | NRL v3 Consumer API       | Source code for the consumer API proxy and specification. | [GitHub repo](https://github.com/NHSDigital/nrl-consumer-api)              |
```
Comment (axelkrastek1-nhs) [inline api/consumer/swagger.yaml:133]: This is missing lambda packaging and lambda logging here and the errors are still below the open source heading.

File: api/consumer/record-locator/consumer.yaml [hunk 6]
```diff
@@ -122,7 +122,22 @@ info:
...
 
+    ### Open source
+
+    You might find the following open source resources useful:
+
+    | Resource                  | Description                                               | Links                                                                      |
+    |---------------------------|-----------------------------------------------------------|----------------------------------------------------------------------------|
+    | NRL v3 API                | Source code for the core API and sandbox                  | [GitHub repo](https://github.com/NHSDigital/nrlf)                          |
+    | NRL v3 Producer API       | Source code for the producer API proxy and specification. | [GitHub repo](https://github.com/NHSDigital/nrl-producer-api)              |
+    | NRL v3 Consumer API       | Source code for the consumer API proxy and specification. | [GitHub repo](https://github.com/NHSDigital/nrl-consumer-api)              |
```
Comment (axelkrastek1-nhs) [inline api/consumer/record-locator/consumer.yaml:133]: This is missing lambda packaging and lambda logging here and the errors are still below the open source heading.

File: api/consumer/swagger.yaml (hunk 5 above)
Comment (nomad3k) [reply api/consumer/swagger.yaml:155]: There was a commit I hadn't pushed.  Should be good now.

File: api/producer/record-locator/producer.yaml (hunk 4 above)
Comment (nomad3k) [reply api/producer/record-locator/producer.yaml]: There was a commit I hadn't pushed.  Should be good now.

File: api/producer/swagger.yaml (hunk 3 above)
Comment (nomad3k) [reply api/producer/swagger.yaml]: There was a commit I hadn't pushed.  Should be good now.

File: swagger/consumer-static/narrative.yaml (hunk 2 above)
Comment (nomad3k) [reply swagger/consumer-static/narrative.yaml:161]: There was a commit I hadn't pushed.  Should be good now.

File: swagger/producer-static/narrative.yaml (hunk 1 above)
Comment (nomad3k) [reply swagger/producer-static/narrative.yaml:150]: There was a commit I hadn't pushed.  Should be good now.

File: api/producer/swagger.yaml [hunk 7]
```diff
@@ -1113,7 +1129,7 @@ components:
             - DocumentReference
         id:
           type: string
-          pattern: "[A-Za-z0-9\\-\\.]{1,64}"
+          pattern: "^(?=.{1,64}$)[A-Za-z0-9\\.]+-[A-Za-z0-9]+[A-Za-z0-9\\_\\-]*$"
```
Comment (axelkrastek1-nhs) [inline api/producer/swagger.yaml:1132]: This should not have been included. Line 1116 is the current state

File: api/producer/record-locator/producer.yaml [hunk 8]
```diff
@@ -1016,7 +1032,7 @@ components:
             - DocumentReference
         id:
           type: string
-          pattern: "[A-Za-z0-9\\-\\.]{1,64}"
+          pattern: "^(?=.{1,64}$)[A-Za-z0-9\\.]+-[A-Za-z0-9]+[A-Za-z0-9\\_\\-]*$"
```
Comment (axelkrastek1-nhs) [inline api/producer/record-locator/producer.yaml:1035]: This should not have been included. Line 1019 is the current state

**Review by axelkrastek1-nhs — APPROVED** [review]

**Pull request merged**